compiler/rustc_codegen_llvm/src/coverageinfo/map_data.rs

```diff
@@ -146,8 +143,14 @@ impl<'tcx> FunctionCoverage<'tcx> {
             self.instance
         );
 
+        let counter_expressions = self.counter_expressions();
+        // Expression IDs are indices into `self.expressions`, and on the LLVM
+        // side they will be treated as indices into `counter_expressions`, so
+        // the two vectors should correspond 1:1.
+        assert_eq!(self.expressions.len(), counter_expressions.len());
```

Comment:
A debug assertion may be sufficient here? With the code right now, this is trivially true, as you are doing a `self.expressions.iter()...collect()`. So this is just a guard so this assumption does not change with any future refactors.

Reply:
Given that it’s trivially true, I don’t see any reason *not* to use a full assertion here, especially since it should get optimised out anyway.

(Note also that debug assertions aren’t checked in `run-coverage` tests by default.)